pflake8/__init__.py

```diff
@@ -51,16 +54,42 @@ class DivertingSafeConfigParser(ConfigParserTomlMixin, configparser.SafeConfigPa
     pass
 
 
-class ModifiedConfigFileFinder(flake8.options.config.ConfigFileFinder):
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.project_filenames = self.project_filenames + ('pyproject.toml',)
-
-
 configparser.RawConfigParser = DivertingRawConfigParser
 configparser.ConfigParser = DivertingConfigParser
 configparser.SafeConfigParser = DivertingSafeConfigParser
-flake8.options.config.ConfigFileFinder = ModifiedConfigFileFinder
+
+
+class FixFilenames(ast.NodeTransformer):
+    tuple_of_interest = ("setup.cfg", "tox.ini", ".flake8")
+
+    def visit_Tuple(self, node: ast.Tuple) -> Any:
+        if all(isinstance(el, ast.Constant) for el in node.elts) and set(
+            self.tuple_of_interest
+        ) == {el.value for el in node.elts}:
+            node.elts.append(ast.Constant(value="pyproject.toml"))
+            ast.fix_missing_locations(node)
+        return node
+
+    @classmethod
+    def apply(cls, module=flake8.options.config):
+        filename = module.__file__
+
+        exec(
+            compile(
+                cls().visit(
+                    ast.parse(
+                        Path(filename).read_text(encoding='UTF-8'), filename=filename
+                    )
+                ),
+                filename=filename,
+                mode="exec",
+            ),
+            module.__dict__,
+        )
+
+
+FixFilenames.apply()
```

Comment:
Could we add some kind of integration test that runs in GitHub actions that validates that `pflake8` is working properly? That would make future breakages easier to detect and PRs easier to remove.